package uk.ac.ed.inf.sdp.group4;

// Generate Services Code
option py_generic_services = true;
option java_generic_services = true;


message WorldObject {
  message Position {
	required int32 x = 1;
	required int32 y = 2;
  }

  message Vector {
	required int32 direction = 1;
	optional float velocity = 2;
  }

  required Position position = 1;
  optional Vector vector = 2;
}

message WorldStateResponse {
  required bool changed = 1;
  required uint64 timestamp = 2;

  optional WorldObject ball = 3;
  optional WorldObject blue = 4;
  optional WorldObject yellow = 5;
}

message WorldStateRequest {
  required uint64 last_timestamp = 1;
}

service WorldStateService {
  rpc GetWorldState(WorldStateRequest) returns (WorldStateResponse);
}
